packages/medusa/src/api-v2/store/customers/middlewares.ts

```diff
@@ -0,0 +1,22 @@
+import { transformBody } from "../../../api/middlewares"
+import { MiddlewareRoute } from "../../../loaders/helpers/routing/types"
+import { StorePostCustomersReq } from "./validators"
+import authenticate from "../../../utils/authenticate-middleware"
+
+export const storeCustomerRoutesMiddlewares: MiddlewareRoute[] = [
+  {
+    method: "ALL",
+    matcher: "/store/customers*",
+    middlewares: [authenticate("store", ["session", "bearer"])],
```

Comment:
Think this middleware is really nice to work with - you can imagine other users picking this up for custom routes:

e.g.:
```
import { authenticate } from "@medusajs/medusa"

// my route stuff
... authenticate("agent", ["session"]) ...
```